src/Peachpie.Library.XmlDom/DOMDocument.cs

```diff
@@ -481,11 +481,14 @@ public virtual DOMNodeList getElementsByTagNameNS(string namespaceUri, string lo
         }
 
         /// <summary>
-        /// Not yet implemented.
+        /// Gets the first element with the matching ID attribute.
         /// </summary>
+        /// <param name="elementId">The attribute ID to match.</param>
+        /// <returns>A <see cref="DOMElement"/>.</returns>
         public virtual DOMElement getElementById(string elementId)
         {
-            throw new NotImplementedException();
+            XmlElement element = XmlDocument.GetElementById(elementId);
+            return new DOMElement(element);
```

Comment:
the function `getElementById` returns `null` if the element is not found

Reply:
@jakubmisek So, `DOMElement(null)` does not return `null`?

Reply:
`new DOMElement` certainly does not :)

Reply:
@jakubmisek Which one would you suggest is the better style?

### A
```cs
return element == null ? element : new DOMElement(element);
```

### B
```cs
return element == null ? null : new DOMElement(element);
```

### C
```cs
return element != null ? new DOMElement(element) : null;
```

Reply:
haha :) `C` is fine - the most frequent branch first

Reply:
@jakubmisek Awesome, never thought of it that way.
I just committed this for review